File: RoyAndRopes.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int t,len,u[1000000],l[1000000],i;
    cin>>t;
    while(t--)
    {
        cin>>len;
        for(i=0;i<len-1;i++)
        cin>>u[i];
        for(i=0;i<len-1;i++)
        cin>>l[i];
        int tim=len,maxi=0;
        for(i=0;i<len-1;i++)
        {
            if(u[i]<(tim-i-1))
            u[i]=0;
            else
            u[i]-=tim-i-1;
            if(l[i]<(tim-i-1))
            l[i]=0;
            else
            l[i]-=tim-i-1;
            maxi=max(maxi,max(l[i],u[i]));
        }
        cout<<maxi+tim<<endl;
    }
}

File: RoyANDRopes.py
t=int(input())
while t>0:
    len=int(input())
    u=[]
    l=[]
    u=str(input()).split()
    l=str(input()).split()
    maxi=0
    for i in range(len-1):
                if int(u[i])<(len-i-1):
                    u[i]='0'
                else:
                    val=int(u[i])-(len-1-i)
                    u[i]=str(val)
                if int(l[i])<(len-i-1):
                    l[i]='0'
                else:
                    val=int(l[i])-(len-1-i)
                    l[i]=str(val)
                maxi=max(maxi,max(int(l[i]),int(u[i])))
    print (maxi+len)
    t-=1
